File: MinhasVacinas/app/src/main/java/com/example/yuriepriscilla/minhasvacinas/RegisterActivity.java
package com.example.yuriepriscilla.minhasvacinas;

import android.content.Intent;
import android.os.Bundle;
import android.os.Build;
import android.os.Bundle;
import android.provider.ContactsContract;
import android.support.v7.app.AppCompatActivity;
import android.text.TextUtils;
import android.view.KeyEvent;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.inputmethod.EditorInfo;
import android.widget.ArrayAdapter;
import android.widget.AutoCompleteTextView;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;


/**
 * Created by Yuri e Priscilla on 09/11/2017.
 */

public class RegisterActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_register);

        final Button buttoncancelregister = findViewById(R.id.registerCancelButton);
        buttoncancelregister.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), LoginActivity.class);
                startActivity(intent);
                finish();
            }
        });

    }

}

File: MinhasVacinas/app/src/main/java/com/betoe/minhasvacinas/RegisterActivity.kt
package com.betoe.minhasvacinas

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_register.*

class RegisterActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register)

        registerSaveButton.setOnClickListener {

            var validator = validateFields()

        }

        registerCancelButton.setOnClickListener {
            var loginIntent = Intent(this, LoginActivity::class.java)
            startActivity(loginIntent)
        }
    }

    

    fun verifyBlankSpaces(): Boolean {
        var errorMessage: String = "Por favor, preencher: "
        val name = registerNameId?.text.toString().trim()
        val email = registerEmailId?.text.toString().trim()
        val confirmEmail = registerConfirmEmailId?.text.toString().trim()
        val password = registerPasswordId?.text.toString().trim()
        val confirmPassword = registerConfirmPasswordId?.text.toString().trim()
        var validator = true

        if (name.isNullOrBlank()) {
            errorMessage = errorMessage + "\nNome"
            validator = false
        }
        if (email.isNullOrBlank()) {
            errorMessage = errorMessage + "\nEmail"
            validator = false
        }
        if (confirmEmail.isNullOrBlank()) {
            errorMessage = errorMessage + "\nConfirmar Email"
            validator = false
        }
        if (password.isNullOrBlank()) {
            errorMessage = errorMessage + "\nSenha"
            validator = false
        }
        if (confirmPassword.isNullOrBlank()) {
            errorMessage = errorMessage + "\nConfirmar Senha"
            validator = false
        }

        if (validator == false) {
            Toast.makeText(this, errorMessage, Toast.LENGTH_LONG).show()
        }

        return validator

    }
}
